### DISCUSSION TITLE
Async select with grouping and only 1 option per group

### DISCUSSION THREAD

**Discussion by cubecleveland** [JedWatson/react-select#4909] [General]
Hi, I am getting great answers from this forum and here is yet another one:

We have a use case of Async select that will allow only 1 selection per group
Right now here is where we are..(
On a a promise return inside loadOption we reorganize the data to include grouping and return a well formatted group data.) how would we take it from here only allowing in the multi select to allow 1 option per group ?

**Comment by ebonow** [comment]
I would approach it by using the `optionIsDisabled` prop and updating the list of selected group options when an onChange event occurs.
